src/Http/Controllers/Api/Geojson/VolumeImagesController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Biigle\Modules\Geo\Http\Controllers\Api\Geojson;
+
+use DB;
+use Biigle\Volume;
+use Biigle\Image;
+use GeoJson\Feature\{Feature, FeatureCollection};
+use GeoJson\Geometry\Point;
+use Biigle\Http\Controllers\Api\Controller;
+use League\Flysystem\FileNotFoundException;
+
+class VolumeImagesController extends Controller
+{
+  public function index($id)
+  {
+    $volume = Volume::findOrFail($id);
+    $this->authorize('access', $volume);
+
+    $images = $volume->images()->select('id', 'lat', 'lng');
+    $labels = $images->join('annotations', 'annotations.image_id', '=', 'images.id')
+      ->join('annotation_labels', 'annotation_labels.annotation_id', '=', 'annotations.id')
+      ->join('labels', 'labels.id', '=', 'annotation_labels.label_id')
+      ->select('images.id', 'images.lat', 'images.lng','labels.name')->get()
+      ->groupBy('id')->map(function($item, $key){
+        return $item->groupBy('name')->map(function($v){
+          return $v->count();
+        });
+      });
+
+    $features = $images->get()->map(function($image) use($labels){
```

Comment:
`$images->get()` will return one entry for each annotation and not for each image, as the `$images` object is internally modified by the subsequent `join` calls. You should rather fetch the images in a separate query and loop over them:

```php
$images = $volume->images()->select('id', 'lat', 'lng')->get();
$labels = $volume->images()->join('annotations', 'annotations.image_id', '=', 'images.id')
   // ...
$features = $images->map(function($image) use ($labels) {
  // ...
```